phpBB/phpbb/textreparser/base.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\textreparser;
+
+abstract class base implements reparser_interface
+{
+	/**
+	* {@inheritdoc}
+	*/
+	abstract public function get_max_id();
+
+	/**
+	* Return all records in given range
+	*
+	* @param  integer $min_id Lower bound
+	* @param  integer $max_id Upper bound
+	* @return array           Array of record
+	*/
+	abstract protected function get_records($min_id, $max_id);
+
+	/**
+	* Add fields to given record, if applicable
+	*
+	* The enable_* fields are not always saved to the database. Sometimes we need to guess their
+	* original value based on the text content or possibly other fields
+	*
+	* @param  array $record Original record
+	* @return array         Complete record
+	*/
+	protected function add_missing_fields(array $record)
+	{
+		if (!isset($record['enable_bbcode'], $record['enable_smilies'], $record['enable_magic_url']))
+		{
+			$record += array(
+				'enable_bbcode'    => $this->guess_bbcodes($record),
+				'enable_smilies'   => $this->guess_smilies($record),
+				'enable_magic_url' => $this->guess_magic_url($record),
+			);
+		}
+
+		// Those BBCodes are disabled based on context and user permissions and that value is never
+		// stored in the database. Here we test whether they were used in the original text.
+		$bbcodes = array('flash', 'img', 'quote', 'url');
+		foreach ($bbcodes as $bbcode)
+		{
+			$field_name = 'enable_' . $bbcode . '_bbcode';
+			$record[$field_name] = $this->guess_bbcode($record, $bbcode);
```

Comment:
you can use the `get_records()` to get the poster id, then you can use this poster_id to get the user permissions through the \phpbb\auth\auth class. In the current state of the perm and auth system it should be possible even if you may have a instantiate a new auth class per user (it should be faster than having to reset the class between each post)